models/sharky/solana/sharky_solana_events.sql

```diff
@@ -35,107 +30,333 @@ WITH sharky_txs AS (
-               WHEN (
-                            startswith(first_ix_data, 'E05D904D3D118936') -- RepayLoan
-                            OR startswith(first_ix_data, 'BB51FA5992571440') -- RepayLoanEscrow
-                        ) THEN 'Repay'
-               WHEN (
-                            startswith(first_ix_data, 'CB5477E28901B417') -- ForecloseLoan
-                            OR startswith(first_ix_data, 'DAF5ED6D2ECE0D0E') -- ForecloseLoanEscrow
-                        ) THEN 'Foreclose'
-               WHEN (
-                            startswith(first_ix_data, '02D0DEBE6D94F775') -- ExtendLoan
-                            OR startswith(first_ix_data, '35BC6DD273220509') -- ExtendLoanEscrow
-                        ) THEN 'Extend'
-            END AS evt_type
-    FROM decoded_events
+), final_event AS (
+    SELECT * FROM offers
+    UNION ALL
+    SELECT * FROM with_amount
 )
-SELECT *,
-       CASE
-           WHEN evt_type = 'Offer' THEN
-               CASE
-                   WHEN block_number > 164178914 THEN sharky_instructions[0].account_arguments[3] -- upgrade to version 3.0 2PXg25v2YGfvFCyX9najJ2mNS5QADvVYNUTtXYYhbyN3cHERqEgbnuDnVKnStaQAMa1edRhM9GAZaHYMPdR9jX8o
-                   ELSE sharky_instructions[0].account_arguments[2]
-               END
-           WHEN evt_type = 'Take' THEN
-               CASE
-                   WHEN block_number > 184280858 THEN sharky_instructions[0].account_arguments[4] -- upgrade to version 5.0 5BjeJ7Wnf7QxHSBEhXN4htn4mECD36SfsEUZbMvzhAF5sMPMCCV1jcu2rigNV8SZohVL7J3ctLLgXZ4gVzzPkmJ7
-                   WHEN block_number > 177070316 THEN sharky_instructions[0].account_arguments[5] -- upgrade to version 4.0 zFA9ffgu3Z1ETH2LdH3o6C6xzKWdJFcgA3qoXo84S6zQ2iJyaRfe61VXw6BBFGM5rFsHFm9fKSUrxEuUTy7grWR
-                   WHEN block_number > 164178914 THEN sharky_instructions[0].account_arguments[6] -- upgrade to version 3.0 2PXg25v2YGfvFCyX9najJ2mNS5QADvVYNUTtXYYhbyN3cHERqEgbnuDnVKnStaQAMa1edRhM9GAZaHYMPdR9jX8o
-                   WHEN block_number > 132405709 THEN sharky_instructions[0].account_arguments[5] -- upgrade to version 2.0 2oNjJSxAM72Y6t7ALKXBiMDjtwhmxEfbk8jBz4es3U6XpBfw2Jvgcu3pSvxuHUkDWbVE48xxpjwgpuxrNnPFjvxm
-                   ELSE sharky_instructions[0].account_arguments[6]
-               END
-           WHEN (evt_type = 'Rescind' OR evt_type = 'Repay' OR evt_type = 'Foreclose' OR evt_type = 'Extend')
-               THEN sharky_instructions[0].account_arguments[0]
-           END as loan_id
-FROM events
-;
\ No newline at end of file
+SELECT
+    'solana' AS blockchain,
+    'sharky' AS project,
+    call_tx_id AS tx_hash,
+    call_block_date AS block_date,
+    CAST(date_trunc('month', call_block_date) AS DATE) AS block_month,
+    call_block_time AS block_time,
+    call_block_slot AS block_number,
+    (amount_raw / 1e9) * p.price AS amount_usd,
+    amount_raw / 1e9 AS amount_original,
+    CAST(amount_raw AS DECIMAL(38, 0)) AS amount_raw,
```

Comment:
similar to question above, but this time it's decimal here?

Reply:
here we can't use `uint256` because `amount_raw` might be negative for example when extending loans with a `new loan amount < old loan amount`